src/lib/features/scheduler/scheduler-service.ts

```diff
@@ -1,19 +1,16 @@
 import { Logger, LogProvider } from '../../logger';
-import MaintenanceService from '../../services/maintenance-service';
+import { IMaintenanceStatus } from '../maintenance/maintenance-service';
 
 export class SchedulerService {
     private intervalIds: NodeJS.Timer[] = [];
 
     private logger: Logger;
 
-    private maintenanceService: MaintenanceService;
+    private maintenanceStatus: IMaintenanceStatus;
 
-    constructor(
-        getLogger: LogProvider,
-        maintenanceService: MaintenanceService,
-    ) {
+    constructor(getLogger: LogProvider, maintenanceStatus: IMaintenanceStatus) {
         this.logger = getLogger('/services/scheduler-service.ts');
-        this.maintenanceService = maintenanceService;
+        this.maintenanceStatus = maintenanceStatus;
```

Comment:
Will this restrict you from using other methods on the maintainance service? If I read it correctly the typechecker would at least complain?